### PR TITLE
Conversion fail for ov::hint::performance_mode with UNDEFINED value

### PR DESCRIPTION
**Pull request by yangwang201911**
Update conversion of ov::hint::PerformanceMode to string.

### Details:
 - Update UNDEFINED value from empty string to "UNDEFINED".
### Tickets:
 - 100498

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by peterchen-intel** [comment]
@yangwang201911 GNAPluginConfigTest.GnaConfigDefaultConfigIsExpected is impacted by this change, please check if we need to update test case behavior or the implementation is incomplete.

**Comment by yangwang201911** [comment]
> @yangwang201911 GNAPluginConfigTest.GnaConfigDefaultConfigIsExpected is impacted by this change, please check if we need to update test case behavior or the implementation is incomplete.

Yes. will updated soon.

**Review by riverlijunjie — APPROVED** [review]

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -640,7 +640,20 @@ QueryNetworkResult MultiDeviceInferencePlugin::QueryNetwork(const CNNNetwork&
...
+        if (workModeAuto) {
+            // set performance hint to 'LATENCY' model for AutoExecutable Network.
+            queryconfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT,
+                                                   PluginConfigParams::LATENCY);
+        } else {
+            // set performance hint to 'THROUGHPUT' model for MultiExecutable Network.
+            queryconfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT,
+                                                   PluginConfigParams::THROUGHPUT);
+        }
+    }
```
Comment (songbell) [inline src/plugins/auto/plugin.cpp:656]: why we have to update this perf hint setting in queryNetwork

Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp]: Removed now.

Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp]: This logic is redundant after initializing performance mode config of AUTO plugin with default value "UNDEFINED" .

File: src/plugins/intel_gna/tests/unit/gna_plugin_config_test.cpp [hunk 2]
```diff
@@ -33,7 +33,7 @@ const std::map<std::string, std::string> supportedConfigKeysWithDefaults = {
     {GNA_CONFIG_KEY(LIB_N_THREADS), "1"},
     {CONFIG_KEY(SINGLE_THREAD), CONFIG_VALUE(YES)},
     {CONFIG_KEY(LOG_LEVEL), PluginConfigParams::LOG_NONE},
-    {CONFIG_KEY(PERFORMANCE_HINT), ""},
+    {CONFIG_KEY(PERFORMANCE_HINT), "UNDEFINED"},
     {CONFIG_KEY(PERFORMANCE_HINT_NUM_REQUESTS), "1"}};
```
Comment (songbell) [inline src/plugins/intel_gna/tests/unit/gna_plugin_config_test.cpp:37]: let's update doc as well.
like here: https://github.com/songbell/openvino/blob/36cb32a8f5d749f4f897721481f670df0370a26a/samples/cpp/benchmark_app/README.md#L186

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**